mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -322,10 +329,11 @@ class LogisticRegression @Since("1.2.0") (
       new BreezeOWLQN[Int, BDV[Double]]($(maxIter), 10, regParamL1Fun, $(tol))
     }
 
-    val initialCoefficientsWithIntercept =
-      Vectors.zeros(if ($(fitIntercept)) numFeatures + 1 else numFeatures)
+    val numFeaturesWithIntercept = if ($(fitIntercept)) numFeatures + 1 else numFeatures
+    val initialCoefficientsWithIntercept = optInitialCoefficients.getOrElse(
+      Vectors.zeros(numFeaturesWithIntercept))
 
```

Comment:
here, 

```scala
val initialCoefficientsWithIntercept =	
   Vectors.zeros(if ($(fitIntercept)) numFeatures + 1 else numFeatures)

if (optInitialModel.isDefined && optInitialModel.get.coefficients.size == numFeatures) {
  val initialCoefficientsWithInterceptArray = initialCoefficientsWithIntercept.toArray
  optInitialModel.get.coefficients.foreachActive { case (index, value) =>
    initialCoefficientsWithInterceptArray(index) = value
  }
  if ($(fitIntercept) {
          initialCoefficientsWithInterceptArray(numFeatures) == optInitialModel.get.intercept
  } 
} else if ($(fitIntercept)) {
      /*
         For binary logistic regression, when we initialize the coefficients as zeros,
         it will converge faster if we initialize the intercept such that
         it follows the distribution of the labels.
         {{{
         P(0) = 1 / (1 + \exp(b)), and
         P(1) = \exp(b) / (1 + \exp(b))
         }}}, hence
         {{{
         b = \log{P(1) / P(0)} = \log{count_1 / count_0}
         }}}
       */
      initialCoefficientsWithIntercept.toArray(numFeatures)
        = math.log(histogram(1) / histogram(0))
    }
```